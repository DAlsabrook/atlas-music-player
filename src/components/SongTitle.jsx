import Skeleton from 'react-loading-skeleton';
import 'react-loading-skeleton/dist/skeleton.css';

export function SongTitle(props) {
    return (
        <div className="flex flex-col my-2">
            {props.loading ? (
                <>
                    <Skeleton className="w-3/4 h-10 mb-2" />
                    <Skeleton className="w-1/2 h-6" />
                </>
            ) : (
                <>
                    <p className='font-bold text-4xl'>Midnight Journey</p>
                    <p className='opacity-50 text-xl my-2'>Echoes of Eternity</p>
                </>
            )}
        </div>
    );
}
