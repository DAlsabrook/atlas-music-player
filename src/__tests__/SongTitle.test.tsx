import { afterEach, expect, test } from "vitest";
import { cleanup, render } from "@testing-library/react";
import { SongTitle } from "@/components/SongTitle.tsx";

// Cleans DOM and any change that may have happened in a test
afterEach(() => {
    cleanup();
});

export type Song = {
  artist: String
  cover: string
  duration: Number
  genre: String
  id: String
  lyrics: String
  song: String
  title: String
}

const fetchPlaylist = async () => {
    const response = await fetch('http://localhost:5173/api/v1/playlist');
    if (response.ok) {
        const data = await response.json();
        const detailedSongs = await Promise.all(data.map(async (song: Song) => {
        const songDetailsResponse = await fetch(`http://localhost:5173/api/v1/songs/${song.id}`);
        if (songDetailsResponse.ok) {
            const songDetails = await songDetailsResponse.json();
            return { ...song, ...songDetails };
        }
        return song;
        }));
        const fullDetailedSongs = await Promise.all(detailedSongs.map(async (song: Song) => {
        const songDetailsResponse = await fetch(`http://localhost:5173/api/v1/lyrics/${song.id}`);
        if (songDetailsResponse.ok) {
            const songDetails = await songDetailsResponse.json();
            return { ...song, ...songDetails };
        }
            return song;
        }));
        return fullDetailedSongs
    }
};

test('Correct inputs', async () => {
    const playlist = await fetchPlaylist();
    if (!playlist) throw new Error('Playlist is undefined');
    const container = render(<SongTitle loading={true} playlist={playlist} currentSong={0}/>)
    expect(container).toMatchSnapshot()
})

test('not loading', async () => {
    const playlist = await fetchPlaylist();
    if (!playlist) throw new Error('Playlist is undefined');
    const container = render(<SongTitle loading={false} playlist={playlist} currentSong={0}/>)
    expect(container).toMatchSnapshot()
})

test('Playlist empty', () => {
    const container = render(<SongTitle loading={true} playlist={[]} currentSong={0}/>)
    expect(container).toMatchSnapshot()
})

test('Current song out of index', async () => {
    const playlist = await fetchPlaylist();
    if (!playlist) throw new Error('Playlist is undefined');
    const container = render(<SongTitle loading={true} playlist={playlist} currentSong={1}/>)
    expect(container).toMatchSnapshot()
})
