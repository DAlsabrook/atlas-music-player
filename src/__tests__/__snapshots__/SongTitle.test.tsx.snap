// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Correct inputs 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col my-2"
      >
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton w-3/4 h-10 mb-2"
          >
            ‌
          </span>
          <br />
        </span>
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton w-1/2 h-6"
          >
            ‌
          </span>
          <br />
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col my-2"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton w-3/4 h-10 mb-2"
        >
          ‌
        </span>
        <br />
      </span>
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton w-1/2 h-6"
        >
          ‌
        </span>
        <br />
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Current song out of index 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col my-2"
      >
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton w-3/4 h-10 mb-2"
          >
            ‌
          </span>
          <br />
        </span>
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton w-1/2 h-6"
          >
            ‌
          </span>
          <br />
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col my-2"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton w-3/4 h-10 mb-2"
        >
          ‌
        </span>
        <br />
      </span>
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton w-1/2 h-6"
        >
          ‌
        </span>
        <br />
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Playlist empty 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col my-2"
      >
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton w-3/4 h-10 mb-2"
          >
            ‌
          </span>
          <br />
        </span>
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton w-1/2 h-6"
          >
            ‌
          </span>
          <br />
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col my-2"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton w-3/4 h-10 mb-2"
        >
          ‌
        </span>
        <br />
      </span>
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton w-1/2 h-6"
        >
          ‌
        </span>
        <br />
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`not loading 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col my-2"
      >
        <p
          class="font-bold text-4xl"
        >
          Starlight Scene
        </p>
        <p
          class="opacity-50 text-xl my-2"
        >
          Dance Troop 5
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col my-2"
    >
      <p
        class="font-bold text-4xl"
      >
        Starlight Scene
      </p>
      <p
        class="opacity-50 text-xl my-2"
      >
        Dance Troop 5
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
